snippets/visualbasic/VS_Snippets_CLR_System/system.io.compression.ziparchive/vb/program1.vb

```diff
@@ -6,16 +6,38 @@ Module Module1
 
     Sub Main()
         Dim zipPath As String = "c:\example\start.zip"
-        Dim extractPath As String = "c:\example\extract"
+
+        Console.WriteLine("Provide path where to extract the zip file:")
+        Dim extractPath As String = Console.ReadLine()
+
+        ' Normalizes the path.
+        extractPath = Path.GetFullPath(extractPath)
+
+        ' Ensures that the last character on the extraction path
+        ' is the directory separator char. 
+        ' Without this, a malicious zip file could try to traverse outside of the expected
+        ' extraction path.
+        If (extractPath(extractPath.Length - 1) <> Path.DirectorySeparatorChar) Then
```

Comment:
The enclosing parens aren't needed here. This should be just:
`If extractPath(extractPath.Length - 1 <> Path.DirectorySeparatorChar Then`